app/src/components/TableOfContents.tsx

```diff
@@ -30,35 +55,31 @@ const TableOfContents = () => {
     { text: string; id: string; level: number }[]
   >([]);
 
-  /** active heading (first in view) */
-  const [active, setActive] = useState("");
+  /** active heading id (first in view) */
+  const [activeId, setActiveId] = useState("");
 
-  /** click off to close on small screens */
-  useClickAway(root, () => {
-    if (window.innerWidth < 1500) setOpen(false);
+  /** click off to close */
```

Comment:
Tweaks to table of contents behavior. Test auto-close functionality on various screen widths. Should close when clicking off on small screens. Should close when it covers an important element on the page for more than a few moments.